src/celeritas/ext/VecgeomParams.cc

```diff
@@ -195,93 +258,21 @@ void VecgeomParams::build_tracking()
     CELER_LOG(status) << "Initializing tracking information";
     ScopedProfiling profile_this{"initialize-vecgeom"};
     ScopedMem record_mem("VecgeomParams.build_tracking");
-    {
-        ScopedTimeAndRedirect time_and_output_("vecgeom::ABBoxManager");
-        vecgeom::ABBoxManager::Instance().InitABBoxesForCompleteGeometry();
-    }
-
-    // Init the bounding volume hierarchy structure
-    vecgeom::cxx::BVHManager::Init();
 
+    // common tracking setup
     if (celeritas::device())
     {
-        // NOTE: this must actually be escaped with preprocessing because the
-        // VecGeom interfaces change depending on the build options.
 #if CELERITAS_USE_CUDA
-        cudaLimit const cuda_attrs[]
-            = {cudaLimitStackSize, cudaLimitMallocHeapSize};
-        char const* const cuda_attr_labels[] = {"stack size", "heap size"};
-        Array<std::size_t, std::size(cuda_attrs)> orig_limits;
-        for (auto i : range(orig_limits.size()))
-        {
-            CELER_CUDA_CALL(cudaDeviceGetLimit(&orig_limits[i], cuda_attrs[i]));
-        }
-
-        auto& cuda_manager = vecgeom::cxx::CudaManager::Instance();
+        CELER_ASSERT(setup_cuda_mem_limits());
```

Comment:
You *cannot* run anything with side effects inside an assertion! This means the cuda mem limits will not be set in release mode.